File: src/langs/java/WheelPart.java
package wheelOfWasm;

import org.teavm.interop.Export;

public class WheelPart {

    public static void main(String[] args) {
        // Required by the compiler
    }

    @Export(name = "name")
    public String name() {
    	return "Java";
    }

    @Export(name = "feelingLucky")
    public int feelingLucky() {
    	return (int)(Math.random() * 100) + 1;
    }
}

File: src/langs/kotlin/wheel-part.kt
import kotlin.math.floor;
import kotlinx.wasm.jsinterop.ReturnSlot_getDouble;

fun main() {
  // Required by the compiler
}

@Retain
public fun name(): String {
  return "Kotlin"
}

@Retain
public fun feelingLucky(): Int {
  // When returning a double from JavaScript, it is put in a global
  // storage as two parts (lower & upper). The way to access this double,
  // is by calling ReturnSlow_getDouble()
  rand();
  return (floor(ReturnSlot_getDouble() * 100) + 1).toInt()
}

@SymbolName("Konan_js_rand")
external public fun rand(): Double